bigquery_etl/cli/query.py

```diff
@@ -1262,6 +1265,151 @@ def render(name, sql_dir, output_dir):
+        if parsed is None:
+            raise click.BadParameter("Format must be <column-name>::<yyyy-mm-dd>")
+        return parsed
+
+
+def _parse_check_output(output: str) -> str:
+    output = output.replace("\n", " ")
+    if "ETL Data Check Failed:" in output:
+        return f"ETL Data Check Failed:{output.split('ETL Data Check Failed:')[1]}"
+    return output
+
+
+@query.command(
+    help="""
+    \b
+    UNDER ACTIVE DEVELOPMENT See https://mozilla-hub.atlassian.net/browse/DENG-919
+
+    Run ETL checks.
+    \b
+
+    Example:
+
+     ./bqetl query check ga_derived.downloads_with_attribution_v2 --partition download_date::2023-05-01
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@sql_dir_option
+@click.option(
+    "--partition",
+    "-p",
+    help="Partition to check, format <column-name>::<yyy-mm-dd>, must be provided if not executing in Airflow ",
+    type=click.UNPROCESSED,
+    callback=_validate_partition_date,
+    required=False,
+)
+@click.pass_context
+def check(ctx, name, sql_dir, partition):
+    """Run a check."""
+    if not is_authenticated():
+        click.echo(
+            "Authentication to GCP required. Run `gcloud auth login` "
+            "and check that the project is set correctly."
+        )
+        sys.exit(1)
+
+    checks_file, project_id, dataset_id, table = paths_matching_checks_pattern(
+        name, sql_dir, project_id=None
+    )
+
+    _check_query(
+        checks_file,
+        project_id,
+        dataset_id,
+        table,
+        partition,
+        ctx.args,
+    )
+
+
+def _check_query(
+    checks_file,
+    project_id,
+    dataset_id,
+    table,
+    partition,
+    query_arguments,
+):
+    """Run the check."""
+    query_arguments.append("--use_legacy_sql=false")
+    if project_id is not None:
+        query_arguments.append(f"--project_id={project_id}")
+
+    # Partition will be None if from Airflow
+    if partition is None:
+        # Need to check the query_arguments for the value
+        for parameter in query_arguments:
+            if parameter.startswith("--parameter"):
+                param_value = parameter.split("=")[1]
+                partition = _parse_partition_date(param_value)
+                # once we have a value that passed the date check stop checking.
+                if partition is not None:
+                    break
+    else:
+        # We have a partition value from the CLI so add to the query_arguments.
+        # There should only be 1
+        key, value = next(iter(partition.items()))
+        query_arguments.append(f"--parameter={key}::{value}")
+
+    if partition is None:
+        raise ValueError("No partition specified to check.")
+
+    jinja_params = {
+        **{"project_id": project_id, "dataset_id": dataset_id, "table_name": table},
+        **partition,
+    }
+
+    with tempfile.NamedTemporaryFile(mode="w+") as query_stream:
```

Comment:
Is there a specific reason why we write the generated template into file vs just keeping it memory and passing it into `bq query` as text?

Reply:
It's how it is currently done in `_run_query`, no reason to not follow the same pattern.